### DISCUSSION TITLE
A `Black` alternative that supports single quotes by default

### DISCUSSION THREAD

**Discussion by jsh9** [jsh9/cercis#28] [General]
Hi @gabroo, @ofek, @hyperknot, @cas--, @davvid, @bofm, @kalekseev, @Kristinita, @roganov, @alanhamlett, @ranjanashish, @bmedicke, @ariddell, @reupen, @lig, @kbd, @ricardo-reis-1970, @skygragon:

I'm tagging you in this discussion, because I saw you left comments on `Black`'s issues to ask for a configurable option to use single quotes rather than double quotes.

Below are the posts that I could find:
- https://github.com/psf/black/issues/1536
- https://github.com/psf/black/issues/594
- https://github.com/psf/black/issues/118
- https://github.com/psf/black/issues/3709
- https://github.com/psf/black/issues/3570

I know it has been as long as 5 years since some of you shared your opinions there, and some of you might have already given up.  But I have not, and I recently forked `Black`, rebranded it to a different name, and defaulted to using single quotes.

Here is the repo: https://github.com/jsh9/cercis

This fork is on top of the latest of `Black`, and I plan to keep rebasing on top of `Black`'s latest changes.

Please give it a try, and please your feedback and/or feature requests are welcome!

**Comment by jsh9** [comment]
Hi @sshishov, I saw your post here (https://github.com/grantjenks/blue/issues/94) and it appeared that you had an issue with `Black`'s default choice of double quotes as well.

Unlike `Blue`, in my project here, I forked `Black` and added configuration options (including the single-quote options).  Therefore, my project doesn't have a flake8 dependency issue.

Please feel free to check it out, and share feedback or feature requests if any!

**Comment by ofek** [comment]
Hello! I'm sorry but I will never use that as I am waiting for Ruff https://github.com/astral-sh/ruff/issues/4798#issuecomment-1579105031

**Comment by sshishov** [comment]
@jsh9 currently we are using  black... we got used to double quotes so far...

At the same time we replaced `flake8` with `ruff` and therefore if `ruff` will implement their own auto-formatter, we will move to that 100%. Therefore I feel that this project is kind of useless nowdays, especially if you cannot keep the same pace as black with all the features and all release cadence...

**Reply by bofm** [reply]
Same for me. It turned out the opinionatedness of Black is very practical. Some may not like the decisions it made, but it makes the code look almost the same on different projects which is more valuable than one's personal preferences. 

Thanks for the effort and mentioning anyway @jsh9.

**Reply by jsh9** [reply]
@sshishov I figured out a structured and repeatable way to keep rebasing on top of the latest `Black`: https://github.com/jsh9/cercis/blob/4b2b93e1fd4844cd22cc931c80e89e7151c51b34/HOW_TO_REBASE.md

I'm glad that I chose to fork instead of doing what `Blue` was doing.  Rebasing is much easier.

**Comment by ranjanashish** [comment]
We are using the following combo to solve this - example pre-commit config file.

```yaml
repos:
  - repo: https://github.com/psf/black
    rev: 22.6.0
    hooks:
      - id: black
        language_version: python3.10
        args: [--skip-string-normalization]

  - repo: https://github.com/pre-commit/pre-commit-hooks
    rev: v4.3.0
    hooks:
      - id: double-quote-string-fixer
```

**Reply by jsh9** [reply]
Yep, this works too.  A caveat would be: your pre-commit return status may never be 0.  In my use cases, the CI pipeline will only pass when all pre-commit steps returns 0.

**Comment by davvid** [comment]
Thanks for creating this. I switched git-cola over to cercis:
- https://github.com/git-cola/git-cola/commit/802ff36decbd5ea2aa7c7856f0a50b10ea7ea054
- https://github.com/git-cola/git-cola/commit/e714f1e265958888b1e3806f3a13420240f10164
